src/cpp/r/session/RSession.cpp

```diff
@@ -285,13 +292,22 @@ void deferredRestoreNewSession()
       r::exec::IgnoreInterruptsScope ignoreInterrupts;
 
       std::string path = string_utils::utf8ToSystem(globalEnvPath.absolutePath());
-      Error error = r::exec::executeSafely(boost::bind(
-                                        R_RestoreGlobalEnvFromFile,
-                                        path.c_str(),
-                                        TRUE));
-      if (error)   
+      
+      std::string errMessage;
+      Error error = restoreGlobalEnvFromFile(path, &errMessage);
+      if (error)
       {
-         reportDeferredDeserializationError(error);
+         LOG_ERROR(error);
```

Comment:
Should this error be shown to the user (as was done previously in `reportDeferredDeserializationError`)?